From feb344d993d8b5b078944d0354d097740e53c011 Mon Sep 17 00:00:00 2001
From: Anson Huang <Anson.Huang@nxp.com>
Date: Fri, 22 Sep 2017 12:06:09 +0800
Subject: [PATCH 2562/5242] MLK-16526-1 thermal: imx_sc: add buffer for
 passive cooling mechanism

commit  1c89dc608d6471ac9ba36d2dcc6528b2f1f453f1 from
https://source.codeaurora.org/external/imx/linux-imx.git

On i.MX8QM/8QXP, When temperature exceeds passive point,
the cooling mechanism will be trigger and temperature
will begin to drop, to avoid back and forth surrounding
the passive point, here adds 10 C buffer for passive point,
that means when cooling mechanism is trigger, only after
the temperature drop to 10 C below the passive point,
the cooling mechanism will exit.

Signed-off-by: Anson Huang <Anson.Huang@nxp.com>
Reviewed-by: Bai Ping <ping.bai@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/thermal/imx_sc_thermal.c |    5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/thermal/imx_sc_thermal.c b/drivers/thermal/imx_sc_thermal.c
index 8503c38..eb4c535 100644
--- a/drivers/thermal/imx_sc_thermal.c
+++ b/drivers/thermal/imx_sc_thermal.c
@@ -20,6 +20,8 @@
 
 #include "thermal_core.h"
 
+#define IMX_SC_TEMP_PASSIVE_COOL_DELTA	10000
+
 struct imx_sc_sensor {
 	struct thermal_zone_device *tzd;
 	struct thermal_cooling_device *cdev;
@@ -91,7 +93,8 @@ static int imx_sc_tsens_get_trend(void *p, int trip, enum thermal_trend *trend)
 	trip_temp = (trip == IMX_TRIP_PASSIVE) ? sensor->temp_passive :
 					     sensor->temp_critical;
 
-	if (sensor->tzd->temperature >= trip_temp)
+	if (sensor->tzd->temperature >=
+		(trip_temp - IMX_SC_TEMP_PASSIVE_COOL_DELTA))
 		*trend = THERMAL_TREND_RAISE_FULL;
 	else
 		*trend = THERMAL_TREND_DROP_FULL;
-- 
1.7.9.5

